core/invoker/src/main/scala/whisk/core/containerpool/docker/DockerContainer.scala

```diff
@@ -98,7 +98,12 @@ object DockerContainer {
     for {
       _ <- pulled
       id <- docker.run(image, args).recoverWith {
-        case _ => Future.failed(WhiskContainerStartupError(s"Failed to run container with image '${image}'."))
+        case BrokenDockerContainer(brokenId, message) =>
+          docker.rm(brokenId)
+          Future.failed(
```

Comment:
Do we want to wait for the remove to be finished or do we not care?

Reply:
I don't think we want to wait. If the removal fails, there is nothing we could do to recover and at this point, the Docker container creation has failed anyway so that we return a failed Future. Added a comment to the code.

A few lines below, we also remove the Docker container if `inspectIPAddress` fails and do not wait for the removal to complete.